tests/lean/linterMissingDocs.lean

```diff
@@ -78,3 +78,18 @@ macro "my_macro" : term => `(my_elab)
 class abbrev BarAbbrev (α : Prop) := Bar α
 
 register_option myOption : Bool := { defValue := my_macro, descr := "hi mom" }
+
+elab (name := myCmd) (docComment)? "my_command" ident : command => pure ()
+
+my_command x
+
+open Lean.Linter.MissingDocs in
+@[missingDocsHandler myCmd]
+def handleMyCmd : SimpleHandler := fun stx => do
+  if stx[0].isNone then
+    lintNamed stx[2] "my_command"
```

Comment:
This wouldn't have happened with a syntax match :)

Reply:
I'm not sure what you mean by "this". The code is correct AFAICT. There are two reasons I used explicit syntax indexing instead of syntax match in most missing docs handlers:

1. The matches were a huge performance issue.
2. Making an exhaustive match of all arguments of a command is tricky to get right, and is not forward compatible with changes (extensions) to the existing syntax. (Of course using indices is also a forward compatibility hazard but there are fewer things that can go wrong.)

Neither of these apply to the test case (or the user code that will presumably be modeled on the test case), so I've changed this one to use syntax match.

Reply:
Sorry for being oblique about the bug, but looks like you found it. Anyway, regarding 1., do you mean the performance issue we noticed in the PR (which would not get any better with syntax match, yeah), or syntax match compared to regular match?

Reply:
Actually I wouldn't say I've found the bug. I noticed that the test case was wrong but the code still seems like it should be correct: the `stx` here should have the form `(docComment)? "my_command" ident` and hence `stx[2]` should be a `Syntax.ident`. The code did seem to find the correct span, but `getId` was returning `.anonymous` and I don't get that. Did we introduce another level of indirection on identifiers?

Regarding syntax match vs regular match, it's true we experienced the issues with regular match but there is no reason to believe syntax match will be better. I took a look at the generated code and maybe it has something to do with the matcher construction? It does a whole bunch of useless stuff like proving that previous cases don't apply. It also does a nested pattern match using `Name.rec` instead of an equality if-else chain like the naive code would, although it does use if-else chains for the underlying string equality checks. Honestly I'm still not totally sure what the cost was; it might be worthwhile to take the previous code and turn it into a test case since it seems to be really stressing something.

Reply:
Oh my, I completely misread the indices! It's actually from `macro/elab`: https://github.com/leanprover/lean4/blob/fbef8a32e19f8d6c473216aaa7ffc0eca6ff20ab/src/Lean/Elab/MacroArgUtil.lean#L42-L43. But it occurs *only* when ignoring a parameter in the actual macro/elab and then later accessing it in a different context like your linter handler, so I'd like to claim that it should not happen for real use cases.